### ISSUE TITLE
Remove `Permitted` Label

### ISSUE THREAD

**Issue by pavlovcik** [ubiquity/ubiquibot#328]
This is my second proposal to dismantle the extraneous label. 

The reason provided for keeping this label as a dependency is to preserve GitHub API rate limits. However, I believe that we can achieve a similar result by filtering through issues that are "closed as completed"

Almost all of the issues that are closed as completed have a bounty associated with them.

**Comment by pavlovcik** [comment]
Request for feedback @Steveantor @rndquu @0xcodercrane

**Comment by Steveantor** [comment]
The rate limit concern was about fetching comments, so issues closed as completed won't work in this context.
It mainly targets the /payout command,
We can label all the paid bounties with the "Permitted (prev: Paid)" as a side quest.

**Comment by rndquu** [comment]
The `Permitted` label is used in the payout command [here](https://github.com/ubiquity/ubiquibot/blob/development/src/handlers/comment/handlers/payout.ts#L24)

It basically preserves github API rate limits because we don't fetch all issue comments (to check whether permit was generated) but simply rely on the label existance.

The `/payout` command basically tries to regenerate a permit URL. This is kind of emergency because normally the bot should generate a permit URL when an issue is closed. So I think we can delete the `Permitted` label because the `/payout` normally should be rarely used although it's ok for me to keep an additional label to preserve github API rate limits.

**Comment by pavlovcik** [comment]
> The `/payout` command basically tries to regenerate a permit URL.

We already just open and close  the issue to force generate a new permit. Why did we add the command for it?

**Comment by rndquu** [comment]
> > The `/payout` command basically tries to regenerate a permit URL.
> 
> We already just open and close the issue to force generate a new permit. Why did we add the command for it?

As far as I remember it was added to mitigate edge case when the bot fails to generate a permit due to internal errors

**Comment by Steveantor** [comment]
The audit page can utilize the label as well to boost the performance (all closed permit issues should be labeled for that to happen which isn't a big deal)

**Comment by rndquu** [comment]
> The audit page can utilize the label as well to boost the performance (all closed permit issues should be labeled for that to happen which isn't a big deal)

How exactly the `Permitted` label can boost performance in the audit page? If the issue is closed then it implies that a permit URL was generated so the label is of no use in this case.

**Comment by pavlovcik** [comment]
So in conclusion, my specification is still valid?

**Comment by Steveantor** [comment]
> > The audit page can utilize the label as well to boost the performance (all closed permit issues should be labeled for that to happen which isn't a big deal)
> 
> How exactly the `Permitted` label can boost performance in the audit page? If the issue is closed then it implies that a permit URL was generated so the label is of no use in this case.

Not necessarily, the old issues and even some newer one's didn't have the permits

**Comment by pavlovcik** [comment]
> Not necessarily, the old issues and even some newer one's didn't have the permits

I see, so this is useful if the repository was started without the use of the bot, and then if the bot was added in later. This could be very useful to preserve rate limits in mature code bases. But it only needs to get through it one time, right?

**Comment by Steveantor** [comment]
> > Not necessarily, the old issues and even some newer one's didn't have the permits
> 
> I see, so this is useful if the repository was started without the use of the bot, and then if the bot was added in later. This could be very useful to preserve rate limits in mature code bases with thousands of closed issues (as completed.) But it only needs to get through it one time, right?

yes

**Comment by rndquu** [comment]
> > > Not necessarily, the old issues and even some newer one's didn't have the permits
> > 
> > 
> > I see, so this is useful if the repository was started without the use of the bot, and then if the bot was added in later. This could be very useful to preserve rate limits in mature code bases with thousands of closed issues (as completed.) But it only needs to get through it one time, right?
> 
> yes

We are using `issue.node_id` as a nonce for permit2 URLs. Perhaps we don't even need to dig into an issue's comments because we can match ethereum transaction with an issue using a nonce. 

Example:
1. Get ethereum transactions from a partner's payment wallet
2. Get all issues
3. For each issue try to match `issue.node_id` (nonce) with ethereum transaction

This way we don't need to get issue comments to find a permit2 URL payload. Hence if we don't request issue comments then we can safely remove the `Permitted` label.
